cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/schedule/AbstractSchedulerService.java

```diff
@@ -90,280 +94,150 @@ protected final void stopScheduler() {
 
   @Override
   public void schedule(Id.Program programId, SchedulableProgramType programType, Schedule schedule) {
-    delegate.schedule(programId, programType, schedule);
+    if (schedule instanceof TimeSchedule) {
+      timeScheduler.schedule(programId, programType, schedule);
+    } else if (schedule instanceof StreamSizeSchedule) {
+      streamSizeScheduler.schedule(programId, programType, schedule);
+    }
+    throw new IllegalStateException("Unhandled type of schedule: " + schedule.getClass());
   }
 
   @Override
   public void schedule(Id.Program programId, SchedulableProgramType programType, Iterable<Schedule> schedules) {
-    delegate.schedule(programId, programType, schedules);
+    Set<Schedule> timeSchedules = Sets.newHashSet();
+    Set<Schedule> streamSizeSchedules = Sets.newHashSet();
+    for (Schedule schedule : schedules) {
+      if (schedule instanceof TimeSchedule) {
+        timeSchedules.add(schedule);
+      } else if (schedule instanceof StreamSizeSchedule) {
+        streamSizeSchedules.add(schedule);
+      } else {
+        throw new IllegalStateException("Unhandled type of schedule: " + schedule.getClass());
+      }
+    }
+    if (!timeSchedules.isEmpty()) {
+      timeScheduler.schedule(programId, programType, timeSchedules);
+    }
+    if (!streamSizeSchedules.isEmpty()) {
+      streamSizeScheduler.schedule(programId, programType, streamSizeSchedules);
+    }
   }
 
   @Override
```

Comment:
You should perhaps talk to the UI guys what would be reasonable to display for a stream based schedule.